lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -1414,6 +1420,53 @@ protected function _validateAndCompleteFieldMapping(array &$mapping)
             throw MappingException::missingFieldName($this->name);
         }
 
+        if (
+            PHP_VERSION_ID >= 70400
+            && isset($this->reflClass)
```

Comment:
Just to understand, do you have a case where `$this->reflClass` is not set? I assume when `DisconnectedMetadataFactory` is used while converting an XML/YML mapping or generating entities?

Reply:
The condition is duplicated below as well, can you factor it out into a private method?

```php
private function isTypedProperty($name) : bool
{
}
```

Reply:
When I remove `isset($this->reflClass)` there are 56 test errors and 3 failures with `Error: Call to a member function hasProperty() on null` message. I have no idea if those tests should be fixed or it is expected behavior and don't know Doctrine internals enough yet to determine if it can occur in real usage.

I can work through those tests and enforce setting `$this->reflClass` if that's desired, but someone more experience would need to take a look for real cases where unset `reflClass` is desired.

Reply:
I have also factored condition as requested

Reply:
It makes sense that it fails for when the `DisconnectedMetadataFactory` is used in case of EntityGenerator and exporter and such things. Of course this creates an inconsistency where the types default to "string" when reflClass does not exist. But there is no way to avoid that at the moment, so lets keep it this way.